source/tutorials/gridfs.txt

```diff
@@ -0,0 +1,212 @@
+file in a single document, GridFS divides a file into parts, or chunks, and
+stores each of those chunks as separate documents.
+
+When you query a GridFS store for a file, the driver reassembles the
+chunks as needed.
+
+The code examples in this guide come from the `GridFSTour.scala
+<{+driver-source-gh+}/blob/master/driver-scala/src/integration/scala/tour/GridFSTour.scala>`__
+file in the driver source code GitHub repository.
+
+Prerequisites
+-------------
+
+You must include the following import statements in your program to run the
+code examples in this guide:
+
+.. code-block:: scala
+
+   import java.nio.ByteBuffer
+   import java.nio.charset.StandardCharsets
+   
+   import org.mongodb.scala._
+   import org.mongodb.scala.bson.BsonObjectId
+   import org.mongodb.scala.gridfs._
+   import org.mongodb.scala.model.Filters
+   
+   import tour.Helpers._
+   
+   import scala.util.Success
+
+.. include:: /includes/obs-note.rst
+
+Connect to a MongoDB Deployment
+-------------------------------
+
+First, connect to a MongoDB deployment and declare and define
+a ``MongoDatabase`` instance.
+
+The following code connects to a standalone
+MongoDB deployment running on ``localhost`` on port ``27017``:
+
+.. code-block:: scala
+
+   val mongoClient: MongoClient = MongoClient()
+
+To learn more about connecting to MongoDB deployments,
+see the :ref:`scala-connect` guide.
+
+Create a GridFS Bucket
+----------------------
+
+GridFS stores files in two collections:
+
+- ``chunks``: stores the file chunks
+- ``files``: stores file metadata
+
+The two collections are in a common bucket and the collection names
+are prefixed with the bucket name.
+
+The driver provides the ``GridFSBucket()`` method to
+create ``GridFSBucket`` instances:
+
+.. code-block:: scala
+
+   val myDatabase = mongoClient.getDatabase("mydb")
+   
+   // Create a gridFSBucket using the default bucket name "fs"
+   val gridFSBucket = GridFSBucket(myDatabase)
+
+You can pass a bucket name to the ``GridFSBucket()`` method:
+
+.. code-block:: scala
+
+   // Create a gridFSBucket with a custom bucket name "files"
+   val gridFSFilesBucket = GridFSBucket(myDatabase, "files")
+
+.. note::
+
+   GridFS automatically creates indexes on the ``files`` and ``chunks``
+   collections when you upload data to the GridFS bucket.
+
+Upload to GridFS
+----------------
+
+The ``GridFSBucket.uploadFromObservable()`` method reads the contents
+of an ``Observable[ByteBuffer]`` and saves it to the ``GridFSBucket`` instance.
+
+You can use the ``GridFSUploadOptions`` type to configure the chunk size
+or include additional metadata.
+
+The following example uploads the contents of a
+``Observable[ByteBuffer]`` into ``GridFSBucket``:
+
+.. code-block:: scala
+
+   // Get the input stream
+   val observableToUploadFrom: Observable[ByteBuffer] = Observable(
+     Seq(ByteBuffer.wrap("MongoDB Tutorial..".getBytes(StandardCharsets.UTF_8)))
```

Comment:
Question:
No change requested, but are the double dots intended?

Reply:
Looks like they are part of the source file - https://github.com/mongodb/mongo-java-driver/blob/bec3def372c9500fbe4d347a4f2414d377679da4/driver-scala/src/integration/scala/tour/GridFSTour.scala#L55-L56

But i will edit it to have no extra punctuation